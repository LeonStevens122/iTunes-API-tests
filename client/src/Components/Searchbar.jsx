// JavaScript source code
// import dependancies and libraries

import Button from "react-bootstrap/Button";
import React from "react";
import Form from "react-bootstrap/Form";

// render a search bar to get user input 
// search bar gets text input from a text form 
// and search criteria from a range of radiobuttons

// button click event calls the search event from the server based on the user input
function Searchbar({ criteria, getResults, search }) {
  return (
    <div>
      <div>
        <Form>
          <Form.Label>Search Term</Form.Label>
          <Form.Control
            required
            type="text"
            placeholder="Enter search term"
            defaultValue=""
            onChange={search}
          />
          <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
        </Form>
        <br />

        <Form onChange={criteria}>
          <Form.Group onChange={criteria} controlId="formGroupRadio">
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="movie"
              label="Movie"
              type="radio"
              id="movie"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="podcast"
              label="Podcast"
              type="radio"
              id="podcast"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="musicVideo"
              label="Music Video"
              type="radio"
              id="musicVideo"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="audiobook"
              label="Audio Book"
              type="radio"
              id="audiobook"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="shortFilm"
              label="Short Film"
              type="radio"
              id="shortFilm"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="tvShow"
              label="TV Show"
              type="radio"
              id="tvShow"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="software"
              label="Software"
              type="radio"
              id="software"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="ebook"
              label="E-Book"
              type="radio"
              id="ebook"
            />
            <Form.Check
              name="radioButtonSet"
              custom
              inline
              value="all"
              label="All"
              type="radio"
              id="all"
            />
          </Form.Group>
        </Form>
        <Button onClick={getResults} variant="primary">
          Search
        </Button>
      </div>
    </div>
  );
}

export default Searchbar;
