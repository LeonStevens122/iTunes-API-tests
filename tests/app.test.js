import { expect } from 'chai';
const request = require('request');
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react'
import chai from "chai";
import chaiJestSnapshot from "chai-jest-snapshot";
import Searchbar from '../client/src/Components/Searchbar';
import Enzyme from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


Enzyme.configure({ adapter: new Adapter() });
chai.use(chaiJestSnapshot);

describe('SearchBar', () => {
    it('should render correctly', () => {
        const output = shallow(
            <Searchbar initialValue="Mock Value"
                criteria="Mock Criteria"
                getResults="Mock Value"
                search="Mock Value" />
        );
        expect(shallowToJson(output)).matchSnapshot();
    });
});

// unit tests and suits

describe('Server Status', function () {

    describe('Main page', function () {
        it('status', function (done) {
            request('http://localhost:3001/', function (error, response, body) {
                expect(response.statusCode).to.equal(200);
                done();
            });
        });

        it('content', function (done) {
            request('http://localhost:3001/', function (error, response, body) {
                expect(body).to.equal('Hello World!');
                done();
            });
        });
    });

});



