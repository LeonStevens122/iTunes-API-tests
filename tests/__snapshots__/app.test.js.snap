// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
<div>
  <div>
    <Form
      inline={false}
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Search Term
      </FormLabel>
      <FormControl
        defaultValue=""
        onChange="Mock Value"
        placeholder="Enter search term"
        required={true}
        type="text"
      />
      <Feedback
        type="valid"
      >
        Looks good!
      </Feedback>
    </Form>
    <br />
    <Form
      inline={false}
      onChange="Mock Criteria"
    >
      <FormGroup
        controlId="formGroupRadio"
        onChange="Mock Criteria"
      >
        <FormCheck
          custom={true}
          disabled={false}
          id="movie"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="Movie"
          name="radioButtonSet"
          title=""
          type="radio"
          value="movie"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="podcast"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="Podcast"
          name="radioButtonSet"
          title=""
          type="radio"
          value="podcast"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="musicVideo"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="Music Video"
          name="radioButtonSet"
          title=""
          type="radio"
          value="musicVideo"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="audiobook"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="Audio Book"
          name="radioButtonSet"
          title=""
          type="radio"
          value="audiobook"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="shortFilm"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="Short Film"
          name="radioButtonSet"
          title=""
          type="radio"
          value="shortFilm"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="tvShow"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="TV Show"
          name="radioButtonSet"
          title=""
          type="radio"
          value="tvShow"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="software"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="Software"
          name="radioButtonSet"
          title=""
          type="radio"
          value="software"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="ebook"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="E-Book"
          name="radioButtonSet"
          title=""
          type="radio"
          value="ebook"
        />
        <FormCheck
          custom={true}
          disabled={false}
          id="all"
          inline={true}
          isInvalid={false}
          isValid={false}
          label="All"
          name="radioButtonSet"
          title=""
          type="radio"
          value="all"
        />
      </FormGroup>
    </Form>
    <Button
      active={false}
      disabled={false}
      onClick="Mock Value"
      type="button"
      variant="primary"
    >
      Search
    </Button>
  </div>
</div>
`;
